### PR TITLE
Set up CI

### PR DESCRIPTION
**Pull request by fivegrant**

### CODE REVIEW

**Comment by fivegrant** [comment]
I haven't decided if I want if I want to add a few more steps:
- black
- isort
- mypy (or pyright?)

Also, I may add more router tests, but that might need to go in a separate PR after this is merged.

**Comment by fivegrant** [comment]
todo list from oct 31 meeting:
- fix settings and environment variables
  - move dbml version to `settings.py`
  - remove env variables for api in docker compose
  - pin postgres version
- break apart config
  - move `settings.py` to top level of `src`
  - create a `schemas` directory for api pydantic schemas
  - recreate `db` directory
- figure out uvicorn reloading
- include black, isort, mypy in precommit hook
- log better
  - add responses types to endpoints
  - (?) add generalized logging functions

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+dbml-builder = "^0.3.0"
+sqlalchemy = "^1.4.42"
+pydantic = "^1.10.2"
+uvicorn = "^0.19.0"
+fastapi = "^0.85.1"
+click = "^8.1.3"
+requests = "^2.28.1"
+psycopg2-binary = "^2.9.4"
+pytest = "^7.2.0"
+pre-commit = "^2.20.0"
```
Comment (mattprintz) [inline pyproject.toml:24]: these two should probably be dev dependencies

Comment (fivegrant) [reply pyproject.toml:24]: Cool, will fix now

Comment (fivegrant) [reply pyproject.toml]: fixed

File: tds/main.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+from tds.db import engine, init_dev_content
+from tds.settings import settings
+
+
+@command()
+@option("--host", default="0.0.0.0", type=str, help="Address for the API")
+@option("--port", default=8000, type=int, help="Port to expose API")
+@option("--dev", default=True, type=bool, help="Set development flag")
+@option("--name", default="full", type=str, help="Name of configured API to use")
+def cli(host: str, port: int, dev: bool, name: str) -> None:
```
[the remark this answers is not in the document]
Comment (fivegrant) [reply tds/main.py:27]: Changed `name` to `config_name`. Does this clear things up?

[the remark this answers is not in the document]
Comment (fivegrant) [reply tds/main.py]: changed to `server_config`

**Review by mattprintz — APPROVED** [review]
Looking nice now. Thanks!

**Pull request merged**